x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/time_scaling.tsx

```diff
@@ -0,0 +1,187 @@
+    selectedOperation.timeScalingMode === 'disabled' ||
+    !hasDateHistogram
+  ) {
+    return null;
+  }
+
+  if (!selectedColumn.timeScale) {
+    return (
+      <EuiText textAlign="right">
+        <EuiSpacer size="s" />
+        <EuiPopover
+          ownFocus
+          button={
+            <EuiButtonEmpty
+              size="xs"
+              iconType="arrowDown"
+              iconSide="right"
+              data-test-subj="indexPattern-time-scaling-popover"
+              onClick={() => {
+                setPopoverOpen(true);
+              }}
+            >
+              {i18n.translate('xpack.lens.indexPattern.timeScale.advancedSettings', {
+                defaultMessage: 'Add advanced options',
+              })}
+            </EuiButtonEmpty>
+          }
+          isOpen={popoverOpen}
+          closePopover={() => {
+            setPopoverOpen(false);
+          }}
+        >
+          <EuiText size="s">
+            <EuiLink
+              data-test-subj="indexPattern-time-scaling-enable"
+              color="text"
+              onClick={() => {
+                setState(
+                  mergeLayer({
+                    state,
+                    layerId,
+                    newLayer: {
+                      ...state.layers[layerId],
+                      columns: {
+                        ...state.layers[layerId].columns,
+                        [columnId]: {
+                          ...selectedColumn,
+                          timeScale: DEFAULT_TIME_SCALE,
+                        },
+                      },
+                    },
+                  })
+                );
+              }}
+            >
+              {i18n.translate('xpack.lens.indexPattern.timeScale.enableTimeScale', {
+                defaultMessage: 'Show as rate',
+              })}
+            </EuiLink>
+          </EuiText>
+        </EuiPopover>
+      </EuiText>
+    );
+  }
+
+  return (
+    <EuiFormRow
+      display="columnCompressed"
+      fullWidth
+      label={i18n.translate('xpack.lens.indexPattern.timeScale.label', {
+        defaultMessage: 'Normalize by unit',
+      })}
+    >
+      <EuiFlexGroup gutterSize="s" alignItems="center">
+        <EuiFlexItem>
+          <EuiSelect
+            compressed
+            options={Object.entries(unitSuffixesLong).map(([unit, text]) => ({
+              value: unit,
+              text,
+            }))}
+            data-test-subj="indexPattern-time-scaling-unit"
+            value={selectedColumn.timeScale}
+            onChange={(e) => {
+              setState(
+                mergeLayer({
+                  state,
+                  layerId,
+                  newLayer: {
+                    ...state.layers[layerId],
+                    columns: {
+                      ...state.layers[layerId].columns,
+                      [columnId]: {
+                        ...selectedColumn,
+                        timeScale: e.target.value as TimeScaleUnit,
```

Comment:
I would prefer that we update the default label every time the time scale is changed. I found it pretty confusing when the axis labels were showing "timestamp per 30 minutes" but my data is "Count of records" instead of "Count of records per second"

Reply:
Good Idea, I will integrate this.